custom_components/ramses_cc/water_heater.py

```diff
@@ -93,145 +185,118 @@ class EvohomeDHW(EvohomeZoneBase, WaterHeaterEntity):
+        return self.rf_entity.temperature
 
     @property
-    def extra_state_attributes(self) -> dict[str, Any]:
+    def extra_state_attributes(self) -> Mapping[str, Any]:
         """Return the integration-specific state attributes."""
-        return {
-            "mode": self._device.mode,
-            **super().extra_state_attributes,
-            "schedule": self._device.schedule,
-            "schedule_version": self._device.schedule_version,
+        return super().extra_state_attributes | {
+            "mode": self.rf_entity.mode,
+            "config": self.rf_entity.mode,
+            "schema": self.rf_entity.schema,
+            "schedule": self.rf_entity.schedule,
+            "schedule_version": self.rf_entity.schedule_version,
         }
 
     @property
     def is_away_mode_on(self) -> bool | None:
         """Return True if away mode is on."""
         try:
-            return self._device.tcs.system_mode[CONF_SYSTEM_MODE] == SystemMode.AWAY
+            return self.rf_entity.tcs.system_mode[SZ_SYSTEM_MODE] == SystemMode.AWAY
         except TypeError:
             return
 
+    @property
+    def name(self) -> str | None:
+        """Return the name of the entity."""
+        return self.rf_entity.name
+
     @property
     def target_temperature(self) -> float | None:
         """Return the temperature we try to reach."""
-        return self._device.setpoint
-
-    @callback
-    def async_handle_dispatch(self, *args: Any) -> None:
-        """Process a service request (system mode) for a controller."""
-        if not args:
-            self.update_ha_state()
-            return
+        return self.rf_entity.setpoint
 
-        payload = args[0]
-        if payload.get(UNIQUE_ID) != self.unique_id:
-            return
-
-        with contextlib.suppress(AttributeError):
-            getattr(self, f"svc_{payload[SERVICE]}")(**dict(payload[DATA]))
-
-    def set_operation_mode(self, operation_mode: str) -> None:
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
         """Set the operating mode of the water heater."""
         active = until = None  # for STATE_AUTO
         if operation_mode == STATE_BOOST:
             active = True
-            until = dt.now() + td(hours=1)
+            until = dt.now() + timedelta(hours=1)
         elif operation_mode == STATE_OFF:
             active = False
         elif operation_mode == STATE_ON:
             active = True
 
-        self.svc_set_dhw_mode(
+        await self.async_set_dhw_mode(
             mode=MODE_HA_TO_EVO[operation_mode], active=active, until=until
         )
 
-    def set_temperature(self, temperature: float = None, **kwargs) -> None:
+    async def async_set_temperature(self, temperature: float = None, **kwargs) -> None:
         """Set the target temperature of the water heater."""
-        self.svc_set_dhw_params(setpoint=temperature)
+        self.async_set_dhw_params(setpoint=temperature)
```

Comment:
why make it `async def` rather than `def`? 

Have a look at `@callback` decorator, I think.

Reply:
Thanks, I'll look into that a bit more closely.